ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -93,25 +98,29 @@ public String getMasterKey(String tenantId) {
     }
 
     @Override
-    public String encrypt(String plainText, String tenantId) {
-        initMasterKey(tenantId);
-        final AwsCrypto crypto = AwsCrypto.builder().withCommitmentPolicy(CommitmentPolicy.RequireEncryptRequireDecrypt).build();
-        JceMasterKey jceMasterKey = createJceMasterKey(tenantId);
-
-        final CryptoResult<byte[], JceMasterKey> encryptResult = crypto
-            .encryptData(jceMasterKey, plainText.getBytes(StandardCharsets.UTF_8));
-        return Base64.getEncoder().encodeToString(encryptResult.getResult());
+    public Future<String> encrypt(String plainText, String tenantId) {
+        return CompletableFuture.supplyAsync(() -> {
+            initMasterKey(tenantId);
+            final AwsCrypto crypto = AwsCrypto.builder().withCommitmentPolicy(CommitmentPolicy.RequireEncryptRequireDecrypt).build();
+            JceMasterKey jceMasterKey = createJceMasterKey(tenantId);
+
+            final CryptoResult<byte[], JceMasterKey> encryptResult = crypto
+                .encryptData(jceMasterKey, plainText.getBytes(StandardCharsets.UTF_8));
+            return Base64.getEncoder().encodeToString(encryptResult.getResult());
+        });
```

Comment:
Here is where using `ForkJoinPool.commonPool()` is actually bad and you really do need a thread pool.